test/meadow/batches_test.exs

```diff
@@ -201,21 +302,61 @@ defmodule Meadow.BatchesTest do
       end)
     end
 
-    test "batch_update/4 handles core metadata" do
+    test "process_batch/1 handles core metadata" do
       query = ~s'{"query":{"term":{"workType.id": "IMAGE"}}}'
+      type = "update"
+      user = "user123"
 
       replace = %{
         visibility: %{id: "OPEN", scheme: "VISIBILITY"}
       }
 
+      attrs = %{
+        query: query,
+        type: type,
+        user: user,
+        replace: Jason.encode!(replace)
+      }
+
       assert Works.list_works() |> length() == 3
 
-      assert {:ok, _result} = Batches.batch_update(query, nil, nil, replace)
+      assert {:ok, batch} = Batches.create_batch(attrs)
+      assert {:ok, _result} = Batches.process_batch(batch)
 
       Works.list_works()
       |> Enum.each(fn work ->
         assert work.visibility.id == "OPEN"
       end)
     end
+
+    test "purge_stalled/1 errors timed out batches stalled in_progress" do
+      replace =
+        Jason.encode!(%{
+          administrative_metadata: %{},
+          descriptive_metadata: %{title: ";sdakjf;alsdkjf;aksjdf"}
+        })
+
+      query =
```

Comment:
Looks like you can yank this `query` variable being set here and just use `attrs` (I saw an unused variable warning in the console)